teuthology/task/kernel.py

```diff
@@ -704,6 +705,25 @@ def wait_for_reboot(ctx, need_install, timeout, distro=False):
         time.sleep(1)
 
 
+def get_version_of_running_kernel(remote):
+    """
+    Get the current running kernel version in a format that can be compared
+    with the output of "rpm -q kernel..."
+    """
+    dist_release = remote.os.name
+    uname_r = remote.sh("uname -r").strip()
+    current = None
+    if dist_release in ['opensuse', 'sle']:
+        # "uname -r" returns              4.12.14-lp151.28.36-default
+        # "rpm -q kernel-default" returns 4.12.14-lp151.28.36.1.x86_64
+        # In order to be able to meaningfully check whether the former
+        # is "in" the latter, we have to chop off the "-default".
+        current = re.sub(r"-default$", "", uname_r)
```

Comment:
I wonder if you can reuse `teuthology.repo_utils.rsstrip` method for this?

Reply:
Why is that better? It's just a simple one-liner regex